### PR TITLE
command/sync: fix delete option with empty target

### LINKED ISSUE #576: Error in `sync` command when using `--delete` flag with 2.1.0 release
Hi 👋

Firstly I want to thank you for this great tool!

After updating from `2.1.0-beta.1` to `2.1.0` we've encountered a strange behavior when `sync`ing a local folder to our S3 buckets. We're not using AWS, but I could reproduce this issue with a LocalStack S3 bucket locally.

Using s5cmd in version `2.1.0-beta.1` syncs a local folder without any issues:
```
❯ ls -l
total 0
-rw-r--r--  1 x  staff  0 21 Jun 07:47 Test1.txt
-rw-r--r--  1 x  staff  0 21 Jun 07:47 Test2.txt
❯ s5cmd-2.1.0-beta version
v2.1.0-beta.1-3e08061
❯ s5cmd-2.1.0-beta ls s3://test-bucket/\*
ERROR "ls s3://test-bucket/*": no object found
❯ s5cmd-2.1.0-beta --stat --log debug sync --delete ./ s3://test-bucket/
cp Test2.txt s3://test-bucket/Test2.txt
cp Test1.txt s3://test-bucket/Test1.txt

Operation	Total	Error	Success
sync		1	0	1
cp		2	0	2

❯ s5cmd-2.1.0-beta ls s3://test-bucket/\*
2023/06/21 05:54:06                 0 Test1.txt
2023/06/21 05:54:06                 0 Test2.txt
❯ s5cmd-2.1.0-beta --stat --log debug sync --delete ./ s3://test-bucket/
DEBUG "sync Test1.txt s3://test-bucket/Test1.txt": object is newer or same age and object size matches
DEBUG "sync Test2.txt s3://test-bucket/Test2.txt": object is newer or same age and object size matches

Operation	Total	Error	Success
sync		1	0	1
```

Running the exact same commands with s5cmd in version `2.1.0` results in errors (`ERROR "rm --raw=true": expected at least 1 object to remove`) when using the `--delete` flag with the `sync` command:
```
❯ ls -l
total 0
-rw-r--r--  1 x  staff  0 21 Jun 07:47 Test1.txt
-rw-r--r--  1 x  staff  0 21 Jun 07:47 Test2.txt
❯ s5cmd version
v2.1.0-3efbbe8
❯ s5cmd ls s3://test-bucket/\*
ERROR "ls s3://test-bucket/*": no object found
❯ s5cmd --stat --log debug sync --delete ./ s3://test-bucket/
ERROR "rm --raw=true": expected at least 1 object to remove
cp Test2.txt s3://test-bucket/Test2.txt
cp Test1.txt s3://test-bucket/Test1.txt

Operation	Total	Error	Success
cp		2	0	2
sync		1	1	0

❯ s5cmd ls s3://test-bucket/\*
2023/06/21 05:49:47                 0  Test1.txt
2023/06/21 05:49:47                 0  Test2.txt
❯ s5cmd --stat --log debug sync --delete ./ s3://test-bucket/
DEBUG "sync Test1.txt s3://test-bucket/Test1.txt": object is newer or same age and object size matches
DEBUG "sync Test2.txt s3://test-bucket/Test2.txt": object is newer or same age and object size matches
ERROR "rm --raw=true": expected at least 1 object to remove

Operation	Total	Error	Success
sync		1	1	0
```

Is there something we need to change in CLI usage after updating to version `2.1.0`? I don't see any changes regarding the `sync` command and `--delete` flag in the release notes. As the `sync` command is now failing we cannot update s5cmd in our CI/CD pipelines.

If you need anything further, feel free to let me know.

Thank you!

### PR DESCRIPTION
**Pull request by seruman**
Fixes #576,
Thanks to @Kucukaslan for pointing it out.

### CODE REVIEW

File: command/sync.go
Comment (Kucukaslan) [inline command/sync.go:1]: Thanks for fixing the problem.

_onlyDest_ is a chan object here (it was a slice in the previous version). So I'm not sure if this check actually does what is desired. Shouldn't we check the length of the dstURLs inside the if block?

File: command/sync.go [hunk 1]
```diff
@@ -460,7 +459,7 @@ func (s Sync) planRun(
 	}()
 
 	// only in destination
-	if s.delete {
+	if s.delete && len(onlyDest) > 0 {
```
Comment (Kucukaslan) [inline command/sync.go:462]: Thanks for fixing the problem.

onlyDest is a chan object here (it was a slice in the previous version). So I'm not sure if this check actually does what is desired. Shouldn't we check the length of the dstURLs inside the if block?

File: e2e/sync_test.go [hunk 2]
```diff
@@ -1130,6 +1130,54 @@ func TestSyncS3BucketToLocalWithDelete(t *testing.T) {
...
+	cmd := s5cmd("sync", "--delete", src+"*", dst)
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Success)
+	stdout := result.Stdout()
+	assertLines(t, stdout, map[int]compareFunc{
+		0: equals(`cp %vcontributing.md %vcontributing.md`, src, dst),
+	}, sortInput(true))
+
+	fmt.Println("stdout", "--", stdout, "--")
```
Comment (igungor) [inline e2e/sync_test.go:1165]: debug line

Comment (seruman) [reply e2e/sync_test.go]: 🤦 removed.

**Review by ilkinulas — APPROVED** [review]

**Review by sonmezonur — APPROVED** [review]

**Pull request merged**